From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Yuki Sireneva <sylfn@disroot.org>
Date: Sun, 24 May 2026 19:11:35 +0300
Subject: [PATCH] (Interface) Video controls for GIFs

This patch is adapted from Kotatogram.
Original author: RadRussianRus <radrussianrus@gmail.com>

diff --git a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
index d6a6a060fd..22d244cf44 100644
--- a/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
+++ b/Telegram/SourceFiles/media/view/media_view_overlay_widget.cpp
@@ -1484,9 +1484,7 @@ QSize OverlayWidget::videoSize() const {
 }
 
 bool OverlayWidget::streamingRequiresControls() const {
-	return !_stories
-		&& _document
-		&& (!_document->isAnimation() || _document->isVideoMessage());
+	return !_stories && _document;
 }
 
 QImage OverlayWidget::videoFrame() const {
